pubspec.yaml

```diff
@@ -0,0 +1,180 @@
+# `dependency_overrides` field in this file. The `any` version constraint is
+# used to indicate that the version of the package is not important, as it is
+# managed by the `DEPS` file.
+
+name: _engine_workspace
+
+# Required for workspace support.
+environment:
+  sdk: ^3.5.0-294.0.dev
+
+# Declare all packages that are part of the workspace.
+workspace:
+  - tools/engine_tool
+
+# Declare all dependencies that are used by one or more packages.
+#
+# A few notes:
+# 1. There is no distinction between "dependencies" and "dev_dependencies";
+#    those notions are for *publishing* packages, not for managing a workspace.
+#    Specific packages in the workspace itself will declare whether they are
+#    dependencies or dev_dependencies, but here it is a union of both.
+#
+# 2. The `any` version constraint is used to indicate that the version of the
+#    package is not important, as it is managed by the `DEPS` file. In other
+#    words, "if the test pass, ship it".
+#
+# 3. The `path` field is used to indicate the location of the package in the
+#    local repository itself. This is to *only* be used for local packages that
+#    are part of the workspace. Third party dependencies are managed by the
+#    `DEPS` file: they get a "any" version constraint and are overridden in the
+#    `dependency_overrides` section below.
+#
+# While not enforced by tooling, try to keep this list in alphabetical order.
+dependencies:
+  args: any
+  async_helper: any
+  engine_build_configs:
+    path: ./tools/pkg/engine_build_configs
+  engine_repo_tools:
+    path: ./tools/pkg/engine_repo_tools
+  expect: any
+  file: any
+  litetest:
+    path: ./testing/litetest
+  logging: any
+  meta: any
+  path: any
+  platform: any
+  process_fakes:
+    path: ./tools/pkg/process_fakes
+  process_runner: any
+  smith: any
+
+# Instructs pub on how to resolve the dependencies that are part of "DEPS".
+#
+# For historic reasons, there are ~3 or so places packages might be located:
+#
+# - `./third_party/pkg/{name}`: for packages vended directly as part of "DEPS".
+#    Usually these are Flutter engine specific packages, i.e. they did not exist
+#    in the Dart vended SDK (the other options below). Typically these originate
+#    from pub (https://pub.dev) and are mirrored into a Google Git repository:
+#    <https://flutter.googlesource.com/>.
+#
+# - `./third_party/dart/pkg/{name}`: for packages that lives *in* the Dart SDK,
+#    which is in turn vendored into the Flutter engine repository. You can see
+#    a full list of available packages here:
+#    <https://github.com/dart-lang/sdk/tree/main/pkg>.
+#
+# - `./third_party/dart/third_party/pkg/{name}`: for packages that are vendored
+#    into the Dart SDK from pub.dev. These are often first-party packages from
+#    the Dart team, but not part of the Dart SDK itself. You can see a full list
+#    of available packages here:
+#    <https://github.com/dart-lang/sdk/blob/main/DEPS>.
+dependency_overrides:
```

Comment:
Why are there extra packages in here that aren't referenced in `dependencies` above?  e.g. `async` is listed here but not above.

Reply:
All _transitive_ dependencies have to also be listed. I can update the comments to make that clear.

(Think of this more like `//third_party/.../BUILD` than a traditional pubspec)